Yafc/Widgets/ImmediateWidgets.cs

```diff
@@ -115,15 +116,9 @@ public static class ImmediateWidgets {
         public static bool BuildInlineObjectList<T>(this ImGui gui, IEnumerable<T> list, IComparer<T> ordering, string header, out T selected, int maxCount = 10,
             Predicate<T> checkMark = null, Func<T, string> extra = null) where T : FactorioObject {
             gui.BuildText(header, Font.subheader);
-            List<T> sortedList = new List<T>(list);
-            sortedList.Sort(ordering ?? DataUtils.DefaultOrdering);
+            var sortedList = ordering == DataUtils.AlreadySortedRecipe ? list.AsEnumerable() : list.OrderBy(e => e, ordering ?? DataUtils.DefaultOrdering);
```

Comment:
I'd suggest expanding this ternary into an `if` -- the space that the ternary saves is not worth the amount of effort that takes to decipher it.

Reply:
Without the ternary you have to declare the variable before the if. This makes it harder to see if the variable is certainly set, as well as not allowing `var` usage anymore.

```C#
IEnumerable<T> sortedList;
if (ordering == DataUtils.AlreadySortedRecipe) {
    sortedList = list.AsEnumerable();
}
else {
    sortedList = list.OrderBy(e => e, ordering ?? DataUtils.DefaultOrdering);
}
```
Myself i prefered the old style, but ok :)

Reply:
> not allowing `var` usage anymore

I think `var` in ternary complicates it even more because now you need to first decipher the ternary, and then decide what on `var` will resolve into based on that.